x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connectors/create_connector/components/manual_configuration.tsx

```diff
@@ -40,9 +59,104 @@ export const ManualConfiguration: React.FC<ManualConfigurationProps> = ({
   const closePopover = () => {
     setPopover(false);
   };
-
+  const { selectedConnector, rawName } = useValues(NewConnectorLogic);
   const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
   const [flyoutContent, setFlyoutContent] = useState<'manual_config' | 'client'>();
+  const getCodeSnippet = (): string => {
+    const connectorInfo: NativeConnector | undefined = selectedConnector?.serviceType
+      ? NATIVE_CONNECTOR_DEFINITIONS[selectedConnector.serviceType]
+      : undefined;
+    if (!connectorInfo) {
+      return '';
+    }
+
+    const dynamicConfigValues = Object.entries(
+      connectorInfo.configuration as ConnectorConfiguration
+    )
+      .map(([key, config]) => {
+        const defaultValue = config ? JSON.stringify(config.value) : null;
+        return `    "${key}": ${defaultValue}`;
+      })
+      .join(',\n');
+    const CONSOLE_SNIPPET = dedent` # Example of how to create a ${connectorInfo?.name} connector using the API
+# This also creates related resources like an index and an API key.
+# This is an alternative to using the UI creation flow.
+
+# 1. Create an index
+PUT /connector-${rawName}
+{
+  "settings": {
+    "index": {
+      "number_of_shards": 3,
+      "number_of_replicas": 2
+    }
+  }
+}  
+# 2. Create a connector
+PUT _connector/${rawName}
+{
+  "name": "My ${connectorInfo?.name} connector",
+  "index_name":  "connector-${rawName}",
+  "service_type": "${selectedConnector?.serviceType}"
+}
+# 3. Create an API key
+POST /_security/api_key
+{
+  "name": "${selectedConnector?.serviceType}-api-key",
```

Comment:
shouldn't it reference connector id/name instead of service type? if we have 3 sharepoint online connectors it's hard to distinguish which api key is associated with which connector (and also below role descriptor field)

Reply:
For now we have been using here the same commitment we agreed on generating default names and artifacts here https://github.com/elastic/search-team/issues/7613 @danajuratoni

Reply:
I think it would make more sense to use `"${rawName}-api-key"`

Reply:
That's true and more accurate. Fixed.